### DISCUSSION TITLE
Feature Request - Fork repository ('ies)

### DISCUSSION THREAD

**Discussion by rafidalhaque** [community/community#8415] [Mobile] [answered]
Please add fork repository option on app 🙂

**Comment by brianlovin** [comment, marked as answer]
Thanks for the feedback @rafidalhaque — this isn't currently on our roadmap, but I'll take a note for the team. Out of curiosity, why are you interested in forking on mobile? Are there follow-up things you would want to do after forking a repo?

**Reply by rafidalhaque** [reply]
Yes, and also I want to save times.

Many times of day I spend on mobile than pc.

Sometimes I saw that, there are a new update in forked repo, which i should merge to fork. 

And It saves a lot of time, rather than fork merging by turning on the computer.

Thanks to this as a note for team 🌹

**Reply by Iron-Ham** [reply]
@rafidalhaque it sounds like this is less about forking a repository to your user and more about being able to update your fork of a repository from upstream? Do I have that right?

**Reply by rafidalhaque** [reply]
> @rafidalhaque it sounds like this is less about forking a repository to your user and more about being able to update your fork of a repository from upstream? Do I have that right?

@Iron-Ham , Yes :smiley:

**Reply by axello** [reply]
For me, sorry for hijacking this thread, it is about forking a repo under your account.

User Story: 
While reading tweets, the fediverse, websites on your mobile device, I see many links to github. Now, when I click it, it opens the github app. But then I have to copy-paste the url, go to my browser, log in to github (again and again), find the repo, fork it.
SURELY there is a better use of my time?
As the app has an intimate relation with my account, sending a "fork this" command is a way friendlier option.

But perhaps github does not want people to fork repos so easily??

**Reply by Iron-Ham** [reply]
Hi @axello 

I'd love more information on why forking is the action you're looking to take in that scenario. Are starred repo lists not a good fit for your workflow? 

Thanks,
Hesham

**Reply by axello** [reply]
Hi Hesham, Starring the repo is often the main option, but is of course a completely different thing. Like retweeting an X or starring an X (or whatever it's called now). When I star it, I support the project by giving it _more cloud_. When I want to **work** on the repo, e.g. I saw a bug I want to fix, or improvement, it is nice to do it in an easy way and not the convoluted way I described above.

**Reply by OffensiveBias-08-145** [reply]
In the cyber world, many POCs for CVEs get posted and "staring" the repository is not always reliable. 

Take the XZ for example. When GH reactively took down the repo, the only people who could research the vulnerability and provide information to the community were those who had Forked the repo to begin with.

A LOT of POCs get shared on twitter, and through the use of the mobile app, forking a repository is usually the first step. 

Given that the app can do everything else including view GH Action runs, the inability to fork a repository is a misstep.

Forking is an important part of using Git and any public source control system.

**Comment by jsoref** [comment]
#3333 asked for this. 

https://twitter.com/github/status/1866883185742622746 claims the feature exists
